src/dataframe/dataframe.jl

```diff
@@ -1410,3 +1410,67 @@ end
 function permutecols!(df::DataFrame, p::AbstractVector{Symbol})
     permutecols!(df, index(df)[p])
 end
+
+##############################################################################
+##
+## Expand & Complete dataframes
+##
+##############################################################################
+
+function _expandhelper(col, iterprod, idx)
+    @inbounds for v in iterprod
+        push!(col, v[idx])
+    end
+end
+
+function expand(df::AbstractDataFrame, indexcols; error::Bool=true, complete::Bool=false, fill=missing, replaceallmissing::Bool=false)
```

Comment:
Why do you think we need `replaceallmissing` kwarg? I think that `fill` is enough. If later someone wants to replace the remaining missing values that were originally in the data frame in non-indexcols then it is easily done.